### ISSUE TITLE
Networkd: does not configure device at startup due to assertion failure

### ISSUE THREAD

**Issue by brandsimon** [systemd/systemd#15126]
**systemd version the issue has been seen with**
> systemd 245 (245-3-arch)

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > Network connection was established successfully

**Unexpected behaviour you saw**
 > Network connection failed

**Steps to reproduce the problem**
If I boot my system, iwctl connects successfully to my wifi, but systemd-networkd does not configure it appropriately, because an assertion fails, after this, the device is in failed mode. Reconnecting via iwctl does not help.
If I restart systemd-networkd, it works perfectly fine, so I think, the problem is related to the boot process. This problem got introdcued with 245, with 244 everything worked fine for me.

Here is a log file:
http://sprunge.us/AdNAdz

My config files:
10-namepolicy.link:
```
[Match]
OriginalName=*

[Link]
NamePolicy=keep kernel database onboard slot path
MACAddressPolicy=random
```

20-wirless.network:
```
[Match]
Name=wlp4s0

[Network]
DHCP=yes
IPv6PrivacyExtensions=yes
IPv6AcceptRA=yes

[DHCPv4]
Anonymize=yes
```

I am not sure if this is relevant, but my iwd main.conf:
```
[General]
UseDefaultInterface=true
AddressRandomization=once
AddressRandomizationRange=full
```

**Comment by yuwata** [comment]
Fix is waiting in #15134.

**Comment by brandsimon** [comment]
@yuwata 
Great, thank you very much for the fast fix.
